### DISCUSSION TITLE
Temperature not updating consistently

### DISCUSSION THREAD

**Discussion by kfchiq** [artisan-roaster-scope/artisan#1142] [Q&A]
I'm running Artisan 2.8.2 on Windows 10.  I have a Perfect Prime TC2100 attached over USB.

If I run the TC2100's Thermometer program I see constant updates with the correct temperature, so I know the thermometer and USB connections are working.

When attached to Artisan and I hit Start, the ET immediately updates to what is displayed on the TC2100, but if I warm the thermocouple with my fingers it stays unchanged.  It does change some over time, by a tenth or two of a degree, but it rarely matches the current temp.

Is there any way to trace what it's receiving, and when?  Any ideas on how I can have a good reading at start, and then random. small changes?

I can't Help->Save Settings because it is grayed out.

Thanks in advance for any ideas.

**Comment by bigl00z3** [comment]
> I can't Help->Save Settings because it is grayed out.

You can't save settings while running. You need to press `OFF` and then it should not be grayed out.

Nico

**Comment by kfchiq** [comment]
[artisan-settings.txt](https://github.com/artisan-roaster-scope/artisan/files/11147568/artisan-settings.txt)

Thanks for that hint.  Here it is.

**Reply by bigl00z3** [reply]
I did not see anything wrong in the settings. From what I see online the Perfect Prime TC2100 seems to be the same as the Mastech 6514 but sold under a different name.

Have you checked if you need to install drivers? And if you did do you have the latest? I saw on the [Mastech webpage](https://mastech-group.com/na/en/product-list/content/8660B5FB63ee) that they have downloads available.

When looking online for the [TC1200 manual](https://m.media-amazon.com/images/I/B10lHgKiXJS.pdf)  I also saw that when connected to usb, it switches from battery to USB power . 
Maybe check on an other USB port to ensure that there enough power. And maybe try with an other usb cable just in case.

Nico

**Comment by kfchiq** [comment]
Thanks very much for verifying my config.  I will revisit the USB drivers.  You may have a point about the port, though it works with the TC2100's app.  I can't test anything right now, but I'll post results when I get a chance.

**Comment by kfchiq** [comment]
You nailed, Nico!  I reinstalled the Silicon Labs driver and now it's working.  Not sure what changed, since it appeared to be the same driver as before, but it's working now.  Many thanks!

**Comment by kfchiq** [comment]
One more detail that might help others in the future.  I now realize that loading USB drivers for ESP32 development broke my Artisan connection.  At this point I don't know exactly what change broke it -- the driver name listed in Device Manager is still the same as before --but it's working.  Seeing as the TC2100 Thermometer.exe works either way, I suspect it's possible for Artisan to work that way too, but that's a nice to have, rather than a bug, IMHO.

**Reply by mgmcderm** [reply]
I have same issue on Frameworks laptop w/ Windows 11. I reset artisan, reinstalled the driver, and behavior is still the same. (Extreme sluggishness in temp readings). The readings are fine w/ the temperature app that comes with the Mastech meter.

Any ideas?

**Comment by kfchiq** [comment]
Re-installing the driver specified in the TC2100 manual did it for me, but I'm running Windows 10.
